magento/magento2
# Don't load datepicker module until it is actually needed

### Description (*)
This PR is a part of my effort to improve Magento 2 frontend performance.

Goal here is to eliminate unnecessary loading of datepicker UI component, which is only needed when the shop has date of birth field enabled.

#### Performance improvement

Tested on Magento 2.4-develop with Luma and sample data. 

- 205 → 202 JS requests (-1.46%).
- 698.40kB → 652.57kB JS transferred (-6.56%).

### Manual testing scenarios (*)

1. Enable date of birth field for customers in admin.
2. Verify that `jquery/ui-modules/datepicker.js` is not loaded on pages that do not use datepicker (e.g. homepage, category, product etc.).
3. Verify that above file is loaded and datepicker works properly on pages that use it (e.g. customer registration form page).

### Contribution checklist (*)


### Resolved issues:
1. [x] resolves magento/magento2#28823: Don't load datepicker module until it is actually needed
